signer/threshold_validator.go

```diff
@@ -351,9 +391,20 @@ func (pv *ThresholdValidator) getExistingBlockSignature(chainID string, block *B
 		round,
 		step,
 	}
-	latestBlock, existingSignature := pv.chainState[chainID].lastSignState.GetFromCache(
+
+	cs, ok := pv.chainState.Load(chainID)
+	if !ok {
+		return nil, stamp, fmt.Errorf("failed to load chain state for %s", chainID)
+	}
+
+	css, ok := cs.(ChainSignState)
+	if !ok {
+		return nil, stamp, fmt.Errorf("expected: (ChainSignState), actual: (%T)", cs)
+	}
```

Comment:
This bit of typecasting is repeated a lot. It may clean things up to abstract to our own type or functions. Go should have a generic sync map by now. :( 

Also, I think it's ok to panic here instead of return an error. If anything other than the type we want is present, something is seriously wrong. Test coverage should immediately flag that panic.

Reply:
refactored out into a mustLoadChainState method